lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -148,22 +181,29 @@ class DecomposeAtenSelectIntOp : public OpRewritePattern<AtenSelectIntOp> {
 
 // Calculates the softmax function on the given `input` tensor. Softmax(x) =
 // exp(x)/sum(exp(x)).
+// To avoid overflow we use the following decomposition rule:
+//     x_max = max(input, dim, keepdim = True)
+//     unnorm = aten.exp(input - x_max)
+//     softmax = unnorm / sum(unnorm, dim, keepdim = True)
 template <typename OpTy>
 static Value getSoftmaxResult(OpTy op, Type resultType,
                               PatternRewriter &rewriter) {
   Location loc = op.getLoc();
   Value dim = op.dim();
   Value self = op.self();
-
-  // exp(x)
-  Value exp = rewriter.create<AtenExpOp>(loc, resultType, self);
-  // sum(exp(x))
-  Value sum =
-      createSumAlongDimension(rewriter, loc, op, exp, dim, /*keepDim=*/true);
+  Value xMax =
+      createMaxAlongDimension(rewriter, loc, op, self, dim, /*keepDim*/ true);
```

Comment:
nit: /*keepDim=*/true
Can you help to run `clang-format` on the entire file? Seems the format of quite a few places are off.

Reply:
Done, thanks for noticing.